suse_migration_services/units/product_setup.py

```diff
@@ -58,29 +59,67 @@ def main():
         products_metadata = os.sep.join(
             [root_path, 'etc', 'products.d']
         )
-        baseproduct = os.sep.join(
-            [products_metadata, 'baseproduct']
-        )
-        if os.path.exists(baseproduct):
+        if os.path.exists(products_metadata):
             log.info('Creating backup of Product data')
             Command.run(
                 [
                     'rsync', '-zav', '--delete', products_metadata + os.sep,
                     '/tmp/products.d.backup/'
                 ]
             )
-            log.info('Updating Base Product to be suitable for migration')
-            xml = ElementTree()
-            xml.parse(baseproduct)
-            register_sections = xml.findall('register')
-            for register in register_sections:
-                target_sections = register.findall('target')
-                for target in target_sections:
-                    register.remove(target)
-            xml.write(
-                baseproduct, encoding="UTF-8", xml_declaration=True
-            )
+        baseproducts = get_baseproduct(products_metadata)
+        if len(baseproducts) != 1:
+            if not baseproducts:
+                message = 'There is no baseproduct'
+            else:
+                message = ('Found multiple product definitions '
+                           'without element <flavor>: \n{0}'
+                           .format('\n'.join(baseproducts)))
+            log.error(message)
+            raise DistMigrationProductSetupException(message)
+
+        log.info('Updating Base Product to be suitable for migration')
+        delete_target_registration(baseproducts[0])
     except Exception as issue:
-        message = 'Base Product update failed with {0}'.format(issue)
+        message = 'Base Product update failed with: {0}'.format(issue)
         log.error(message)
         raise DistMigrationProductSetupException(message)
+
+
+def get_baseproduct(products_metadata):
+    prod_filenames = glob.glob(
+        os.path.join(products_metadata, '*.prod')
+    )
+    base_product_files = []
+    xml = ElementTree()
+    for prod_filename in prod_filenames:
+        try:
+            xml.parse(prod_filename)
+            register_sections = xml.findall('register')
+            for register in register_sections:
+                flavor = register.findall('flavor')
+                if not flavor:
+                    base_product_files.append(prod_filename)
+
+        except Exception:
+            log.error('Could not parse {0} as XML file'.format(prod_filename))
```

Comment:
minor issue, we should report the "why" for the exception. Could you change this to be:

```python
except Exception as issue:
    log.error(
        'Parsing XML file {0} failed with: {1}'.format(prod_filename, issue)
    )
```